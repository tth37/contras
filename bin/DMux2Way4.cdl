import And4 from "And4.cdl"
import Not from "Not.cdl"

@Head
NAME DMux2Way4
INPUT a, b, D
OUTPUT Y3, Y2, Y1, Y0
SYMBOL And4 And41, And42, And43, And44
SYMBOL Not Not1, Not2

@Body
Not1(a=a)=>(out=na)
Not2(a=b)=>(out=nb)
And41(a[0]=D, a[1]=D, a[2]=na, a[3]=nb)=>(out=W0)
And42(a[0]=D, a[1]=D, a[2]=a, a[3]=nb)=>(out=W1)
And43(a[0]=D, a[1]=D, a[2]=na, a[3]=b)=>(out=W2)
And44(a[0]=D, a[1]=D, a[2]=a, a[3]=b)=>(out=W3)

@End
OUTPUT Y0=W0,Y1=W1, Y2=W2, Y3=W3