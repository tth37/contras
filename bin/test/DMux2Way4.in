@Case
a=00000010110101
b=01000101111010
D=11111111111111
Y0=?
Y1=?
Y2=?
Y3=?
@End