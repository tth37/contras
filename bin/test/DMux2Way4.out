@Case
 a[0] = 00000010110101
 b[0] = 01000101111010
 D[0] = 11111111111111
Y0[0] = 10111000000000
Y1[0] = 00000010000101
Y2[0] = 01000101001010
Y3[0] = 00000000110000
